cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,366 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
+    endforeach()
+    foreach(gui_plugin_path ${gui_plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in 
+      "prepend-non-duplicate;IGN_GUI_PLUGIN_PATH;@CMAKE_INSTALL_PREFIX@/${gui_plugin_path}\n")
+    endforeach()
+    foreach(variable_export_path ${variable_export_paths})
+      string(REPLACE "," ";" variable_export_path ${variable_export_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in
+      "prepend-non-duplicate;${variable_export_path}\n")
+    endforeach()
+    file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/colcon.pkg "{\n  \"hooks\": [\"share/${CMAKE_PROJECT_NAME}/hooks/hook.dsv\"]\n}")
+
+    configure_file(
+      "hooks/hook.dsv.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/hooks/hook.dsv" @ONLY
+    )
+    install(DIRECTORY
+    ${CMAKE_CURRENT_BINARY_DIR}/hooks
+    DESTINATION share/${PROJECT_NAME})
+
+  elseif(${export_type} STREQUAL "plain-cmake")
+
+    foreach(resource_path ${resources_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in
+      "export IGN_GAZEBO_RESOURCE_PATH=@CMAKE_INSTALL_PREFIX@/share/@PROJECT_NAME@/${resource_path}\n")
+    endforeach()
+    foreach(plugin_path ${plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in 
+      "export IGN_GAZEBO_SYSTEM_PLUGIN_PATH=@CMAKE_INSTALL_PREFIX@/${plugin_path}\n")
+    endforeach()
+    foreach(gui_plugin_path ${gui_plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in 
+      "export IGN_GUI_PLUGIN_PATH=@CMAKE_INSTALL_PREFIX@/${gui_plugin_path}\n")
+    endforeach()
+    foreach(variable_export_path ${variable_export_paths})
+      string(REPLACE "," "=" variable_export_path ${variable_export_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in
+      "export ${variable_export_path}\n")
+    endforeach()
+    file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/export_path.sh.in "#!/usr/bin/env bash\nsource @CMAKE_INSTALL_PREFIX@/share/@PROJECT_NAME@/setup.sh\n")
+    
+    configure_file(
+      "setup.sh.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/share/setup.sh" @ONLY
+    )
+
+    install(FILES
+      ${CMAKE_CURRENT_BINARY_DIR}/share/setup.sh
+      DESTINATION share/${PROJECT_NAME}
+    )
+    
+    configure_file(
+      "export_path.sh.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/share/export_path" @ONLY
+    )
+
+    install(PROGRAMS
+      ${CMAKE_CURRENT_BINARY_DIR}/share/export_path
+      DESTINATION bin
+    )
+
+  else()
+    message(FATAL_ERROR
+    "Unknown export type")
+  endif()
+  
+
+endmacro()
+#####################################################
+#ign_export_variable(variable_name variable_path)
+#Exports path of mentioned varaible 
+
+macro(ign_export_variable variable_name variable_path)
+  list(APPEND variable_export_paths "${variable_name},${variable_path}")
+endmacro()
+
+#######################################################
+#ign_add_plugins(path_to_plugins)
```

Comment:
Follow the format you see in `ign_find_package` to document this. What are the arguments? Which arguments do you pick? What is meant by "COMMON_PUBLIC_LIBRARIES". Same with all your cmake macros.